pkg/agent/entrypoint.go

```diff
@@ -79,18 +75,32 @@ func (r MainRunner) Run(logger *zap.Logger, ctx context.Context) error {
 		}
 	}
 
-	logger.Info("Entering main loop")
-	for {
-		event, err := vmEventQueue.Wait(ctx)
-		if err != nil {
-			if ctx.Err() != nil {
-				// treat context canceled as a "normal" exit (because it is)
-				return nil
-			}
+	mc, err := billing.NewMetricsCollector(ctx, logger, &r.Config.Billing)
+	if err != nil {
+		return fmt.Errorf("error starting billing metrics collector: %w", err)
```

Comment:
This is no longer "starting" the billing metrics collector, right?

Reply:
Yup, changed to creating.